xarray/core/dataset.py

```diff
@@ -1223,7 +1223,8 @@ def rename(self, name_dict, inplace=False):
         ----------
         name_dict : dict-like
             Dictionary whose keys are current variable or dimension names and
-            whose values are new names.
+            whose values are new names (although the new name is permitted
```

Comment:
Maybe it would be better to simple state "are desired names". It seems pretty logical that the same names would be OK.